### ISSUE TITLE
How to exclude One-Hot from gradient propagation

### ISSUE THREAD

**Issue by frankibem** [Microsoft/CNTK#2179]
In Double DQN, I'm trying to compute the value of the next states (post_state_var) using the online model. To vectorize my solution, I've made use of the one_hot op. However, I get the following error when I try to train:
> The node "OneHot" can be used in training, but it does not participate  in gradient propagation.

I've defined my model and inputs as:
```
state_var = cntk.input_variable(state_shape, name='state')
action_var = cntk.input_variable(1, name='action')
reward_var = cntk.input_variable(1, name='reward')
post_state_var = cntk.input_variable(state_shape, name='post_state')
terminal_var = cntk.input_variable(1, name='terminal')

with cntk.default_options(activation=relu):
    model_fn = Sequential([
        Dense(32, name='h1'),
        Dense(32, name='h2'),
        Dense(action_shape, name='action')
    ])

model = model_fn(state_var)
target_model = model.clone(cntk.CloneMethod.freeze)
```
I then calculate the target values and define the loss as follows:
```
# Value of action selected at state t
state_value = cntk.reduce_sum(model * one_hot(action_var, num_classes=action_shape), axis=1)

# Double Q learning - Value of action selected at state t+1
online_post_state_model = model_fn(post_state_var)
online_post_state_best_action = cntk.argmax(online_post_state_model)
post_state_best_value = cntk.reduce_sum(target_model * 
                                        one_hot(online_post_state_best_action, num_classes=action_shape))

gamma = 0.99
target = reward_var + (1.0 - terminal_var) * gamma * post_state_best_value

# MSE for simplicity
td_error = state_value - cntk.stop_gradient(target)
loss = cntk.reduce_mean(cntk.square(td_error))
```

If I replace
`online_post_state_model = model_fn(post_state_var)`
with 
`online_post_state_model = model_fn.clone(cntk.CloneMethod.freeze)(post_state_var)`

then the error is gone, but that's wrong since it's using an old frozen model to compute the target. How can I evaluate `model_fn` with `post_state_var` and exclude the output from back propagation? Am I not using `stop_gradient` correctly?
